lisa/nic.py

```
@@ -0,0 +1,221 @@
+        default_interface_name = default_interface.stdout.strip()
+        assert_that(default_interface_name in self._nic_names).described_as(
+            (
+                f"ERROR: NIC name found as default {default_interface_name} "
+                f"was not in original list of nics {repr(self._nic_names)}."
+            )
+        ).is_true()
+        return default_interface_name
+
+    def _get_host_if_info(self) -> None:
+        # for parsing ip addr show (ipv4)
+        _ip_regex = (
+            r"inet\s+([0-9a-fA-F]{1,3}\.[0-9a-fA-F]{1,3}\."
+            r"[0-9a-fA-F]{1,3}\.[0-9a-fA-F]{1,3})"
+        )
+        _mac_regex = (
+            r"ether\s+([0-9a-fA-F]{2}:[0-9a-fA-F]{1,3}:"
+            r"[0-9a-fA-F]{2}:[0-9a-fA-F]{2}:"
+            r"[0-9a-fA-F]{2}:[0-9a-fA-F]{2})"
+        )
+
+        for nic in self.get_upper_nics():
+            # get ip and mac
+            result = self._node.execute(f"/sbin/ip addr show {nic}", shell=True)
+            result.assert_exit_code()
+            ip_match = re.search(_ip_regex, result.stdout)
+            mac_match = re.search(_mac_regex, result.stdout)
+
+            if ip_match and mac_match:
+                # check we found matches for both
+                for match in [ip_match, mac_match]:
+                    assert_that(len(match.groups())).described_as(
+                        (
+                            f"(IP) Trouble parsing `ip addr show {nic}` output."
+                            " Number of match groups was unexpected."
+                        )
+                    ).is_equal_to(1)
+
+                ip_addr = ip_match.group(1)
+                mac_addr = mac_match.group(1)
+
+                # double check IP address looks right
+                try:
+                    ipaddress.ip_address(ip_addr)
+                except ValueError:
+                    assert_that(False).described_as(
+                        f"{ip_addr} was not recognized as an IP Address."
```

The raw exception may provide the raw text of wrong ip address.. Let it raise Exception, it doesn't need the extra try/except. because it doesn't provide more information.